server/src/main/java/org/elasticsearch/action/admin/indices/template/get/TransportGetComponentTemplateAction.java

```diff
@@ -71,22 +71,21 @@ protected void masterOperation(Task task, GetComponentTemplateAction.Request req
         Map<String, ComponentTemplate> allTemplates = state.metadata().componentTemplates();
 
         // If we did not ask for a specific name, then we return all templates
-        if (request.names().length == 0) {
+        if (request.name() == null) {
             listener.onResponse(new GetComponentTemplateAction.Response(allTemplates));
             return;
         }
 
         final Map<String, ComponentTemplate> results = new HashMap<>();
-        for (String name : request.names()) {
-            if (Regex.isSimpleMatchPattern(name)) {
-                for (Map.Entry<String, ComponentTemplate> entry : allTemplates.entrySet()) {
-                    if (Regex.simpleMatch(name, entry.getKey())) {
-                        results.put(entry.getKey(), entry.getValue());
-                    }
+        String name = request.name();
+        if (Regex.isSimpleMatchPattern(name)) {
+            for (Map.Entry<String, ComponentTemplate> entry : allTemplates.entrySet()) {
+                if (Regex.simpleMatch(name, entry.getKey())) {
+                    results.put(entry.getKey(), entry.getValue());
                 }
-            } else if (allTemplates.containsKey(name)) {
-                results.put(name, allTemplates.get(name));
             }
+        } else if (allTemplates.containsKey(name)) {
+            results.put(name, allTemplates.get(name));
```

Comment:
I think in case when the template is missing here (when asking for a specific template by name) we want to throw a resource not found exception. Like was done here: https://github.com/elastic/elasticsearch/pull/54530/files#diff-e753c14c84c5edb785bc197306e5a24fR180